tests/utils/test_helpers.py

```diff
@@ -66,3 +70,51 @@ def test_get_http_basic_auth_without_password(config):
 
 def test_get_http_basic_auth_missing(config):
     assert get_http_basic_auth(config, "foo") is None
+
+
+def test_parse_author_simple_name_and_email():
```

Comment:
Please write tests in a function named `test_parse_author` and separate cases with comments.

It's good to organize code and tests with the same structure.

Reply:
I do not think this is a good idea.  Each of the eight `test_parse_author_*` functions that I added represents a semantically self-contained test.

Imagine the implementation of `parse_author()` was changed in a way that would cause four of these tests to fail.  If all the assertions were packed together in a single test function, running the tests would reveal only the first one of these failures.

Therefore, if it is merely for the purpose of test organization, I would rather group the related tests into a separate class, but still keep the distinct test functions:

```python
class TestParseAuthor(object):
    """Tests for the :func:`parse_author` function."""

    def test_simple_name_and_email(self):
        [...]

    def test_simple_name_only(self):
        [...]
```

Of course I am not the one to make the calls here. If you prefer having everything in a single `test_parse_author()` function, so be it—just let me know. =)

Reply:
I still think those cases had better be in a function. We can sort out those cases by precedence and treat them as all-or-none cases.

The cases which are not formal as documented should be placed at last.

Reply:
Then, so be it—commit 619593c5ef645285285267844f1ca52445df934f refactors all the tests for `parse_author()` into a single function.